src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -976,8 +976,9 @@ public void Sort(IComparer<T> comparer)
             }
             Contract.EndContractBlock();
 
-            if( _size > 0) {
+            if (_size > 0) {
                 ArraySortHelper<T>.Sort(_items, 0, _size, comparison);
+                _version++;
```

Comment:
The method above increments the version just after the sort method call. So I did the same. It makes no sense to indicate that the collections was changed if it really wasn't.

IMO ideally both methods should check the `_size` or `count` and the condition should be `if (_size >= 2)`  or  `if (count >= 2)` respectively, because sorting a collection that contains 0 or 1 items cannot change it.
If you want, I could add these updated conditions.